### PR TITLE
Arcgis data inside one context to get rid of redundant requests

### PR DESCRIPTION
**Pull request by PavelFilo**
- added ArcGIS context
- texts 
- default offence store by role

### CODE REVIEW

File: state/ArcgisStore/ArcgisStoreProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'core-js/stable/atob'
```
Comment (JurajKubrican) [inline state/ArcgisStore/ArcgisStoreProvider.tsx:1]: might be worth it leaving a comment as to what this relates to

File: modules/map/hooks/useCameraChangeHandler.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ type Dependencies = {
   isMapPinShown: boolean
   selectedPolygon: UdrZoneFeature | null
   setSelectedPolygon: Dispatch<SetStateAction<UdrZoneFeature | null>>
-  setIsMapPinShown: Dispatch<SetStateAction<boolean>>
+  setIsMapPinShown?: Dispatch<SetStateAction<boolean>>
```
[the remark this answers is not in the document]
Comment (PavelFilo) [reply modules/map/hooks/useCameraChangeHandler.ts:21]: this is actually necessary for the next PR (number 25... locations), I accidentally pushed it to this branch

File: app/(app)/scan/scan-result.tsx [hunk 3]
```diff
@@ -3,12 +3,36 @@ import { router, useLocalSearchParams } from 'expo-router'
...
+import { useSnackbar } from '@/components/screen-layout/Snackbar/useSnackbar'
 import { ScanResultEnum } from '@/modules/backend/openapi-generated'
 
 type ScanResultSearchParams = {
   scanResult: ScanResultEnum
 }
 
+const scanResultTextsMap: Record<ScanResultEnum, Record<string, string>> = {
+  [ScanResultEnum.NoViolation]: {
+    title: 'Bez priestupku',
```
Comment (JurajKubrican) [inline app/(app)/scan/scan-result.tsx:15]: I'd seriously start consideringadding i18n as soon as possible as the number of texts is rapidly increasing

File: app/(app)/scan/scan-result.tsx [hunk 4]
```diff
@@ -23,24 +47,34 @@ const getResultVariant = (result?: ScanResultEnum) => {
 // TODO - texts
 const ScanResultPage = () => {
   const { scanResult } = useLocalSearchParams<ScanResultSearchParams>()
+  const snackbar = useSnackbar()
+
+  if (!scanResult) {
+    snackbar.show('Nepodarilo sa načítať výsledok skenovania')
```
Comment (JurajKubrican) [inline app/(app)/scan/scan-result.tsx:53]: I'm not sure what's our strategy for deling with these generic errors
I'd consider adding an error handler that would both show a toast and log an error somewhere
this is definitely outside of scope of this PR

File: components/map/MapCamera.tsx [hunk 5]
```diff
@@ -42,6 +42,8 @@ const MapCamera = forwardRef<Camera, Props>(({ cameraZoom, flyToCenter, setFlyTo
     } else if (!isInitialized) {
       setIsInitialized(true)
       setLocalFlyToCenter(MAP_CENTER)
+    } else if (!flyToCenter) {
+      setLocalFlyToCenter(null)
```
Comment (JurajKubrican) [inline components/map/MapCamera.tsx:46]:
```suggestion
      setLocalFlyToCenter(undefined)
```
If you set it to undefined you don't have to change the state types and the defaults below
although I do see how setting something to null might seem better  Idon't thin kthere's a real diferrence in this case

File: state/ArcgisStore/ArcgisStoreProvider.tsx (hunk 1 above)
Comment (PavelFilo) [reply state/ArcgisStore/ArcgisStoreProvider.tsx:1]: the import is redundant there... will remove

File: app/(app)/scan/scan-result.tsx (hunk 3 above)
Comment (PavelFilo) [reply app/(app)/scan/scan-result.tsx:15]: it's on my agenda after next week's presentations

File: app/(app)/scan/scan-result.tsx (hunk 4 above)
Comment (PavelFilo) [reply app/(app)/scan/scan-result.tsx:53]: i have made changes in PR 27 (added error screen) so I will replace it after merging that PR

File: app/(app)/scan/scan-result.tsx (hunk 3 above)
Comment (JurajKubrican) [reply app/(app)/scan/scan-result.tsx:15]: cc @radoslavzeman as we discussed this offline

**Pull request merged**